### PR TITLE
Updated .NET for late 2023

### PR DESCRIPTION
**Pull request by technosophos**
@itowlson I am WAAAY overdue on making these changes.

### CODE REVIEW

File: c-sharp.md [hunk 1]
```diff
@@ -5,70 +5,37 @@ tags = ["c#", "csharp", "dotnet", ".net", "webassembly"]
...
+last_modified = "2023-10-26T00:50:50Z"
 ---
 # C# and .NET in WebAssembly
 
 The .NET (aka dotnet) ecosystem supports a variety of languages, including C# (C sharp) and ASP.NET. Programs that target .NET are compiled to an intermediary bytecode format that runs in the .NET Common Language Runtime (CLR). This makes .NET an example of a [virtual machine language](/blog/scripts-vs.compiled-wasm.md).
 
 ## Available Implementations
 
-The .NET ecosystem has long had support for browser-side WebAssembly via the [Blazor toolkit](https://dotnet.microsoft.com/en-us/apps/aspnet/web-apps/blazor).
+The .NET ecosystem has long had support for browser-side WebAssembly via the [Blazor toolkit](https://dotnet.microsoft.com/en-us/apps/aspnet/web-apps/blazor). As of .NET 7, it has support for WebAssembly as a compile target.
```
Comment (itowlson) [inline c-sharp.md:16]: Nit: I think .NET 7 may only have been Steve Sanderson's preview. Not 100% sure though. It's definitely in .NET 8.

File: c-sharp.md [hunk 2]
```diff
@@ -5,70 +5,37 @@ tags = ["c#", "csharp", "dotnet", ".net", "webassembly"]
...
 
-The team at Microsoft is rapidly building out a number of WebAssembly tools. In 2022, it seems reasonable to expect that Microsoft will have new tooling for WebAssembly. C# compiled using the .NET toolchain will have access to these WebAssembly features.
-
-At the time of this writing, the closest we have seen to a full WASI runtime is Steve Sanderson's example [.NET WASI Runtime](https://github.com/SteveSandersonMS/dotnet-wasi-runtime).
+The team at Microsoft is rapidly building out a number of WebAssembly tools. .NET 7 includes support for compiling to WebAssembly. Read [Ivan Towlson's excellent tutorial](https://www.fermyon.com/blog/dotnet-wasi) on building a WebAssembly app with .NET.
```
Comment (itowlson) [inline c-sharp.md:18]: The instructions in that are, I think, quite out of date with the .NET 8 changes.

Somebody should write an updated post.  And you should be unreasonably nice about that one too.

File: c-sharp.md [hunk 3]
```diff
@@ -5,70 +5,37 @@ tags = ["c#", "csharp", "dotnet", ".net", "webassembly"]
...
-- The .NET WASI Runtime looks excellent
-
-For the most part, though, it is too early to speculate on the rest of the .NET tooling.
-
 ## Example
 
-Our example uses the [.NET WASI Runtime](https://github.com/SteveSandersonMS/dotnet-wasi-runtime). This is a preview and requires the supporting tools and build steps described in that repository.
-
->> All of our examples follow [a documented pattern using common tools](/wasm-languages/about-examples).
+You will need to be running .NET 7 and also have the [WASI SDK](https://github.com/SteveSandersonMS/dotnet-wasi-sdk).
```
Comment (itowlson) [inline c-sharp.md:24]: That's deprecated with .NET 8.

File: c-sharp.md [hunk 4]
```diff
@@ -5,70 +5,37 @@ tags = ["c#", "csharp", "dotnet", ".net", "webassembly"]
...
-
-	<ItemGroup>
-		<ProjectReference Include="path\to\src\Wasi.Sdk\Wasi.Sdk.csproj" ReferenceOutputAssembly="false" />
-	</ItemGroup>
-
-	<Import Project="path\to\Wasi.Sdk\build\Wasi.Sdk.targets" />
-
-</Project>
-```
+Create a new project with `dotnet new console`. Then add the WASI SDK like this: `dotnet add package Wasi.Sdk --prerelease`
```
Comment (itowlson) [inline c-sharp.md:28]: I think this is now `dotnet workload install wasi-experimental` (and then you run `dotnet new wasiconsole`) but I have kind of lost track of it all.  @jpflueger is likely more up-to-date than me at this point!

File: c-sharp.md [hunk 5]
```diff
@@ -77,39 +44,44 @@ To compile this, run the .NET 7 command line:
...
 Hello, World
 ```
 
+To build more advanced applications, use the [Spin SDK for .NET](https://github.com/fermyon/spin-dotnet-sdk#fast-startup-using-wizer). This [blog post walks you through a new Spin .NET app](https://www.fermyon.com/blog/webassembly-for-dotnet-developers-spin-sdk-intro)
+
 ## Learn More
 
 Here are some great resources:
-
+- The best way to build Spin .NET apps is with the [Spin .NET SDK](https://github.com/fermyon/spin-dotnet-sdk#fast-startup-using-wizer)
```
Comment (itowlson) [inline c-sharp.md:79]: Filip's blog seems like a good and concrete one for this, and pretty recent https://www.strathweb.com/2023/09/dotnet-wasi-applications-in-net-8-0/

**Review by itowlson — APPROVED** [review]
I left a lot of fretful comments about .NET 8 but I am not super confident on the status of that.  It may be a better pointer than Steve's toolkit all the same...?

**Comment by technosophos** [comment]
After talking with Ivan, we decided we'd take a deeper look after KubeCon when we both have some time to do more research

File: c-sharp.md (hunk 1 above)
Comment (jpflueger) [reply c-sharp.md:16]: That's the way I understand it as well. The wasi target is enabled by installing the `wasi-experimental` workload.

File: c-sharp.md (hunk 4 above)
Comment (jpflueger) [reply c-sharp.md:28]: That is spot on. [This repo from Steve is a good reference](https://github.com/SteveSandersonMS/spiderlightning-dotnet) for how it dotnet 8 will work, though our approach is slightly different unless we wanted to follow a similar approach to how go adds handlers during the init function.

**Comment by vdice** [comment]
@technosophos and @itowlson is this PR eligible for an update?  If so, can it be ported to https://github.com/fermyon/developer where the wasm languages guide content now lives?  Thank you.

**Comment by vdice** [comment]
Closing in anticipation of archiving this repo.  Please re-open at the wasm language guide's new home at https://github.com/fermyon/developer/tree/main/content/wasm-languages when convenient.  Thank you!

**Pull request closed without merging**